Document using xpack.ml.use_auto_machine_memory_percent setting in docker getting started (#114009)
elastic/elasticsearch: docs/reference/setup/install/docker.asciidoc

TIP: Use the `-m` flag to set a memory limit for the container. This removes the
need to <<docker-set-heap-size,manually set the JVM size>>.
+

> Machine Learning features such as semantic search with the ELSER model

Reviewer: ```suggestion
Machine learning features such as <<semantic-search-elser, semantic search with ELSER>>
```

> Machine learning features such as <<semantic-search-elser, semantic search with ELSER>>

Reviewer 2: ```suggestion
{ml-cap} features such as <<semantic-search-elser, semantic search with ELSER>>
```

require a larger container with more than 1GB of memory.

> If you intend to use the ML capabilities start the container with this command

Reviewer: ```suggestion
If you intend to use the machine learning capabilities, then start the container with this command:
```

> [source,sh,subs="attributes"]

Reviewer 2: ```suggestion
+
[source,sh,subs="attributes"]
```

----
docker run --name es01 --net elastic -p 9200:9200 -it -m 6GB -e "xpack.ml.use_auto_machine_memory_percent=true" {docker-image}

> ----

Reviewer 2: ```suggestion
----
+
```
